arduino/KnexElevator/DataAcquisitionTask.cpp

```diff
@@ -0,0 +1,117 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+// /////////////////////////////////////////////////////////////////////////////
+
+
+// /////////////////////////////////////////////////////////////////////////////
+// Include(s)
+// /////////////////////////////////////////////////////////////////////////////
+#include <Arduino_FreeRTOS.h>
+#include <Arduino.h>
+
+#include "DataAcquisitionTask.h"
+
+
+// /////////////////////////////////////////////////////////////////////////////
+// Local Types
+// /////////////////////////////////////////////////////////////////////////////
+
+
+////////////////////////////////////////////////////////////////////////////////
+DataAcquisitionTask::DataAcquisitionTask(void)
+{
+    // Intentionally left empty
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+DataAcquisitionTask::~DataAcquisitionTask(void)
+{
+    // Intentionally left empty
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+void
+DataAcquisitionTask::Initialize(void)
+{
+    Serial.print("ENTER: ");
+    Serial.println(__PRETTY_FUNCTION__);
+
+    // Fire up our task
+    xTaskCreate(DataAcquisitionTask::RunWrapper,  // Entry Point
+                "DataAcquisitionTask",            // Task Name
+                128,                              // Stack Depth
+                this,                             // Task Parameters
+                2,                                // Priority
+                NULL);                            // Storage for task handle
+
+    Serial.print("LEAVE: ");
+    Serial.println(__PRETTY_FUNCTION__);
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+void
+DataAcquisitionTask::RunWrapper(void * pvParameters)
+{
+    // Entry point to the new task
+    DataAcquisitionTask* thisTask =
+        reinterpret_cast<DataAcquisitionTask*>(pvParameters);
+    thisTask->Run();
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+void
+DataAcquisitionTask::Run(void)
+{
+    Serial.print("ENTER: ");
+    Serial.println(__PRETTY_FUNCTION__);
+
+    while (true)
+    {
+        char timeStampStringBuf[15+1];
```

Comment:
Probably no need as the safe snprintf() is used which will add a null terminator after the dersired number of characters is written (else bufsize-1).